docs/data-companion/quick-start.md

```diff
@@ -0,0 +1,269 @@
+}
+
+for {
+    select {
+    case <- ctx.Done():
+        return
+    case latestHeight, ok := <-stream:
+        if ok {
+            if latestHeight.Error != nil {
+                // Do something with error
+            } else {
+                // Do something with latestHeight.Height
+            }
+        } else {
+            return
+        }
+    }
+}
+```
+
+The ability to monitor new blocks is truly intriguing as it unlocks avenues for creating dynamic pipelines and ingesting
+services via the producer-consumer pattern. For instance, upon receiving a notification about a fresh block, one can
+activate a method to retrieve block data and save it in a database. Subsequently, the node can be configured to retain
+a specific height, allowing for pruning of the data.
+
+## Storing the fetched data
+
+In the Data Companion workflow, the second step involves saving the data retrieved from a blockchain onto an external
+storage medium, such as a database. This external storage medium is important because it allows the data to be accessed
+and utilized by custom web services that can serve the blockchain data in a more efficient way.
+
+When choosing a database, evaluate your specific needs, including data size, user access, and budget.
+For example, the [RPC Companion](https://github.com/cometbft/rpc-companion) uses Postgresql as a starting point, but there
+are many other options to consider. Choose a database that meets your needs and helps you achieve your objectives.
+
+Before proceeding to the next step, it is crucial to verify that the data has been correctly stored in the external database.
+Once you have confirmed that the data has been successfully stored externally, and if you have evidence that the previous
+data has also been stored externally, you can proceed to update the new "retain_height" information. This update will
+allow the node to prune the information that is now stored externally.
+
+## Pruning the node data
+
+In order to successfully execute the Data Companion workflow, the third step entails utilizing the newly introduced
+gRPC APIs to set the retain height value on the node. The pruning service allows the data companion to effectively influence
+pruning on the node.
+
+One important concept that can affect the pruning of nodes is the `retain_height`. The retain_height determines the specific
+height from which the data can be safely deleted from the node's storage. By considering the retain_height,
+nodes can effectively manage their storage usage and ensure that they are only retaining the data that is necessary for
+their operations. This is important because storage space is a finite resource and nodes with limited storage space may
+struggle to keep up with the growth of the blockchain.
+
+For a comprehensive understanding of the pruning service for the data companion, please see the document
+[Using a Data Companion to influence data pruning on a CometBFT node](./pruning.md).
```

Comment:
I think the diagram you once had on the workflow (which might be in the ADR even could be useful. It can be even simplified, just to visually show the steps described above.

Reply:
The diagram is from ADR102, I can link this in the future for a better reference